src/odemis/driver/ephemeron.py

```diff
@@ -0,0 +1,788 @@
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the
+GNU General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis.
+If not, see http://www.gnu.org/licenses/.
+'''
+# this driver is developed for communication with an EBIC controller API
+# the EBIC scan controller from Ephemeron will acquire a digital EBIC signal
+# to be used simultaneously with other signals such as SE/CL
+
+import logging
+import os
+import queue
+import threading
+import time
+import weakref
+from typing import Any, Optional
+
+import asyncio
+from asyncua import Client, Server, Node, ua
+from asyncua.common.methods import uamethod
+from asyncua.common.statemachine import State, StateMachine, Transition
+import numpy
+from scipy import ndimage
+
+import functions
+# from constants import *
+from odemis import model, dataio
+from odemis.model import Detector, oneway
+from odemis.util import img
+
+# constants
+URI = "opc.tcp://localhost:4840/freeopcua/server/"
+NAMESPACE = "http://examples.freeopcua.github.io"
+DATA_VAR_NAME = "MightyEBICDataArray"
+
+STATE_ID_IDLE = "server-idle"
+STATE_ID_RUNNING = "server-running"
+STATE_ID_TRIGGER = "server-triggered"
+STATE_ID_CHECKING_DWELL = "server-checking-dwell"
+STATE_ID_STOPPED = "server-stopped"
+STATE_ID_ERROR = "server-error"
+
+STATE_NAME_IDLE = "Idle"
+STATE_NAME_RUNNING = "Running"
+STATE_NAME_TRIGGER = "Trigger"
+STATE_NAME_CHECKING_DWELL = "CheckingDwell"
+STATE_NAME_STOPPED = "Stopped"
+STATE_NAME_ERROR = "Error"
+
+TRANSITION_ID_IDLE = "to-idle"
+TRANSITION_ID_RUNNING = "to-running"
+TRANSITION_ID_TRIGGER = "to-trigger"
+TRANSITION_ID_CHECKING_DWELL = "to-checking-dwell"
+TRANSITION_ID_STOPPED = "to-stopped"
+TRANSITION_ID_ERROR = "to-error"
+
+TRANSITION_NAME_IDLE = "To Idle"
+TRANSITION_NAME_RUNNING = "To Running"
+TRANSITION_NAME_TRIGGER = "To Trigger"
+TRANSITION_NAME_CHECKING_DWELL = "To Checking Dwell"
+TRANSITION_NAME_STOPPED = "To Stopped"
+TRANSITION_NAME_ERROR = "To Error"
+
+
+class MightyEbic(Detector):
+    def __init__(self, name: str, role: str, device: str, port: int,
+                 channel: int, url: str, namespace: str, **kwargs):
+        """
+        Initialise the EBIC controller
+        :param name:
+        :param role:
+        :param device:
+        :param port:
+        :param channel:
+        :param url:
+        :param namespace:
+        """
+        super().__init__(name, role, **kwargs)
+        # for debug, eventually -> get resolution from the controller/detector
+        max_res = (1024, 1024)
+        min_res = (1, 1)
+
+        self.device = device
+        self.port = port
```

Comment:
What are device and port? I think you can drop them, and just use the "url" + "namespace" arguments.